inbox/api/jsc_api.py

```diff
@@ -118,3 +120,32 @@ def auth_callback():
         })
 
         return make_response( (resp, 201, { 'Content-Type': 'application/json' }))
+
+@app.route('/retry_sync', methods=['POST'])
+def retry_sync():
+    g.parser.add_argument('namespace_public_id', required=True, type=valid_public_id, location='form')
+
+    args = strict_parse_args(g.parser, request.args)
+    shard = (args.get('target') or 0) >> 48
+
+    with session_scope(shard) as db_session:
+        try:
+            namespace = db_session.query(Namespace) \
+                .filter(Namespace.public_id == args['namespace_public_id']).first()
+
+            if namespace is None:
+                resp = simplejson.dumps({ 'message': 'Namespace do not exists', 'type': 'custom_api_error' })
+                return make_response((resp, 400, { 'Content-Type': 'application/json' }))
+
+            account = namespace.account
+            resp = simplejson.dumps({
+                'error': account.sync_status['sync_error']
+            })
+            account.sync_state = 'running'
+            account.sync_should_run = True
```

Comment:
There are more steps needed here:
We have to enable the credentials and also set the sync_host (which should be grabber from process_id -- take a look on `mailsync/service.py` to see how it works or just ping me)